File: library/src/main/java/me/shkschneider/skeleton/ui/AutoImageViewHeight.java
package me.shkschneider.skeleton.ui;

import android.content.Context;
import android.graphics.drawable.Drawable;
import android.support.v7.widget.AppCompatImageView;
import android.util.AttributeSet;

// <http://stackoverflow.com/a/12283909>
public class AutoImageViewHeight extends AppCompatImageView {

    private float mRatio = 1.0F;

    public AutoImageViewHeight(final Context context) {
        super(context);
    }

    public AutoImageViewHeight(final Context context, final AttributeSet attrs) {
        super(context, attrs);
    }

    public AutoImageViewHeight(final Context context, final AttributeSet attrs, final int defStyleAttr) {
        super(context, attrs, defStyleAttr);
    }

    public float getRatio() {
        return mRatio;
    }

    public void setRatio(final float ratio) {
        mRatio = ratio;
    }

    @Override
    protected void onMeasure(final int widthMeasureSpec, final int heightMeasureSpec) {
        final Drawable drawable = getDrawable();
        if (drawable == null) {
            super.onMeasure(widthMeasureSpec, heightMeasureSpec);
            return;
        }
        final int height = MeasureSpec.getSize(heightMeasureSpec);
        int width = (int) Math.ceil((float) height * (float) drawable.getIntrinsicHeight() / (float) drawable.getIntrinsicWidth());
        setMeasuredDimension((int) (mRatio * width), (int) (mRatio * height));
    }

}

File: library/src/main/kotlin/me/shkschneider/skeleton/ui/AutoImageViewHeight.kt
package me.shkschneider.skeleton.ui

import android.content.Context
import android.support.v7.widget.AppCompatImageView
import android.util.AttributeSet
import android.view.View

// <http://stackoverflow.com/a/12283909>
class AutoImageViewHeight : AppCompatImageView {

    var ratio = 1.1.toFloat()

    constructor(context: Context, attrs: AttributeSet? = null, defStyleAttr: Int = 0) : super(context, attrs, defStyleAttr)

    override fun onMeasure(widthMeasureSpec: Int, heightMeasureSpec: Int) {
        drawable?.let {
            val height = View.MeasureSpec.getSize(heightMeasureSpec)
            val width = Math.ceil((height.toFloat() * drawable.intrinsicHeight.toFloat() / drawable.intrinsicWidth.toFloat()).toDouble()).toInt()
            setMeasuredDimension((ratio * width).toInt(), (ratio * height).toInt())
        } ?: run {
            super.onMeasure(widthMeasureSpec, heightMeasureSpec)
        }
    }

}
